packages/grafana-ui/src/components/Cascader/Cascader.tsx

```diff
@@ -113,7 +115,7 @@ export class Cascader extends PureComponent<CascaderProps, CascaderState> {
     for (const option of searchableOptions) {
       const optionPath = option.value || [];
 
-      if (optionPath.indexOf(initValue) === optionPath.length - 1) {
+      if (optionPath[optionPath.length - 1] === initValue) {
```

Comment:
@tskarhed 2 small fixes here. This could have failed to select option if main and child option have the same value. Seems easier to just test the leaf and compare that.

Also the separator had spaces around when used for custom values vs for other values so unified that.